etcdserver/api/v3rpc/key.go

```diff
@@ -55,6 +55,12 @@ func (s *kvServer) Range(ctx context.Context, r *pb.RangeRequest) (*pb.RangeResp
 	if resp.Header == nil {
 		plog.Panic("unexpected nil resp.Header")
 	}
+
+	if r.KeysOnly {
```

Comment:
sorry. one more little thing. we probably need to move this change one layer inner since txn might have range request inside. So we need to clear out the value at apply phase at line https://github.com/coreos/etcd/blob/master/etcdserver/apply.go#L273